### ISSUE TITLE
dcm2niix from matlab command using variables

### ISSUE THREAD

**Issue by Monsterosaurus** [rordenlab/dcm2niix#229]
Hi,

first of all, thank you so much for the software, it's a great tool.
However, I'm having trouble calling dcm2niix from the command line in MATLAB. 
When I do, and specify all paths it needs to use, it works fine. But I would rather define the paths as variables up front, and call dcm2niix using those variables. 
I have worked it out that is finds the path to dcm2niix.exe, and the path to the output folder, but when I call  the system to execute the function it only gives me an example output filename. 

My code is something like this:
output_folder = 'path\to\my folder\with\spaces\';
input_file = 'path\to\PAR file\myfile.PAR';
d2n = 'path\to executable\with\spaces\dcm2niix.exe';
cmd = ['"',d2n,'" ' ,'-f %p%s -p n -b n -o ' ,'"', output_folder, '" < ' ,  '"' input_file '"'];

and I get the following result:
Chris Rorden's dcm2niiX version v1.0.20180622 (JP2:OpenJPEG) (JP-LS:CharLS) MSC1900 (64-bit Windows) 
Example output filename: '\MPRAGE1.nii' 

ans =

     0

Is there something I'm missing in the command line?

Kind regards,
Monique

**Comment by neurolabusc** [comment]
You need to remove the `<` symbol. Since you are using Matlab you may also want to consider the terrific [dicm2nii](https://github.com/xiangruili/dicm2nii). 

```
output_folder = 'path\to\my folder\with\spaces';
input_file = 'path\to\PAR file\myfile.PAR';
d2n = 'path\to executable\with\spaces\dcm2niix.exe';
cmd = ['"',d2n,'" ' ,'-f %p%s -p n -b n -o ' ,'"', output_folder, '"' , '"' input_file '"'];

```

**Comment by Monsterosaurus** [comment]
Yes I tried tht, but got the same result. dicm2nii works as well, but I cannot specify the same output parameters that I can with dcm2niix, because we are working with PAR REC files where the precision scaling needs to be off.

**Comment by neurolabusc** [comment]
I think the issue is that you have to remove the `<` and you also have to have a space between the " that terminates the input folder and theone that describes the output folder (so they are separated). The code below works for me. So you need  `'" '`. The script

```
output_folder = 'C:\out folder';
input_file = 'C:\in folder\CHRISANTT1W_51.PAR';
d2n = 'C:\dcm2niix folder\dcm2niix.exe';
cmd = ['"',d2n,'" ' ,'-f %p%s -p n -b n -o ' ,'"', output_folder, '" ' , '"' input_file '"'];
system(cmd)
```

correctly creates a file in the specified folder:

```
"C:\dcm2niix folder\dcm2niix.exe" -f %p%s -p n -b n -o "C:\out folder" < "C:\in folder\CHRISANTT1W_51.PAR"

Chris Rorden's dcm2niiX version v1.0.20180622 (JP2:OpenJPEG) (JP-LS:CharLS) MSC1900 (64-bit Windows) 
...
Convert 1 DICOM as C:\out folder\T1W_IR5_ph (512x512x26x1) 
Conversion required 0.516000 seconds. 
```

I tend to like using sprintf to combine strings with Matlab, as it reads easier to my eyes. You just need to remember to use `%%` when you want to print `%`. Here is your script with sprintf:

```
output_folder = 'C:\out folder';
input_file = 'C:\in folder\CHRISANTT1W_51.PAR';
d2n = 'C:\dcm2niix folder\dcm2niix.exe';
cmd = sprintf('"%s" -f %%p%%s -p n -b n -o "%s" "%s"', d2n, output_folder, input_file );
system(cmd)

```
